tc-discord-webhooks-server/src/main/java/com/github/playerforcehd/tcdiscordwebhooks/notificator/DiscordNotificator.java

```diff
@@ -235,13 +236,27 @@ public void notifyBuildSuccessful(@NotNull SRunningBuild sRunningBuild, @NotNull
     @Override
     public void notifyBuildFailed(@NotNull SRunningBuild sRunningBuild, @NotNull Set<SUser> users) {
         String title = "Build failed";
-        String description = "The build with the ID " + sRunningBuild.getBuildId() + " has failed!";
-        String url = this.sBuildServer.getRootUrl() + "/viewLog.html?buildId=" + sRunningBuild.getBuildId();
+        StringBuilder description = new StringBuilder("The build with the ID " + sRunningBuild.getBuildNumber() + " has failed!");
+
+        description.append("\n" + "The build has failed with the following reason: ");
+        // Iterate through the failure reasons of the running build
+
+        // Append the failure reason to the description
+        for (BuildProblemData failureReason : sRunningBuild.getFailureReasons())
+            description.append("\n").append(failureReason.getDescription());
+
+        // Check the length of the description
+        if (description.length() > 1800) {
+            // Truncate the description if it is too long
+            description = new StringBuilder(description.substring(0, 1800));
+        }
+
+        String url = this.sBuildServer.getRootUrl() + "/viewLog.html?buildId=" + sRunningBuild.getBuildNumber();
```

Comment:
Please continue to use `sRunningBuild.getBuildId();` here.
The URL expects the build id. If the build number is customized in the settings, the URL won't work anymore.